spec/integration/apo_spec.rb

```diff
@@ -36,38 +55,106 @@ def user_stub
+      visit url_for(:controller => :apo, :action => :register, :id => @druid)
+      expect(find_field('title'      ).value).to eq('APO Title')
+      expect(find_field('copyright'  ).value).to eq('Copyright statement')
+      expect(find_field('use'        ).value).to eq('Use statement')
+      expect(find_field('managers'   ).value).to eq('dlss:developers')
+      expect(find_field('viewers'    ).value).to eq('sunetid:someone')
+      expect(find_field('desc_md'    ).value).to eq('TEI')
+      expect(find_field('use_license').value).to eq('by-sa')
+      expect(page).to have_no_field('collection')
+    end
+
+    it 'should edit an existing apo' do
+      @druid = 'druid:ab987cd6543'
+      create_apo @druid
+      visit url_for(:controller => :apo, :action => :register, :id => @druid)
+      fill_in 'managers',  :with => 'dlss:developers dlss:psm-staff'
+      fill_in 'viewers',   :with => 'sunetid:someone'
+      fill_in 'title',     :with => 'New APO Title'
+      fill_in 'copyright', :with => 'New copyright statement'
+      fill_in 'use',       :with => 'New use statement'
+      fill_in 'managers',  :with => 'dlss:dpg-staff'
+      fill_in 'viewers',   :with => 'sunetid:anyone'
+      page.select('Attribution No Derivatives 3.0 Unported', :from => 'use_license')
+      page.select('MODS', :from => 'desc_md')
+      click_button 'register'
+      visit url_for(:controller => :apo, :action => :register, :id => @druid)
+      expect(find_field('title'      ).value).to eq('New APO Title')
+      expect(find_field('copyright'  ).value).to eq('New copyright statement')
+      expect(find_field('use'        ).value).to eq('New use statement')
+      expect(find_field('managers'   ).value).to eq('dlss:dpg-staff')
+      expect(find_field('viewers'    ).value).to eq('sunetid:anyone')
+      expect(find_field('desc_md'    ).value).to eq('MODS')
+      expect(find_field('use_license').value).to eq('by-nd')
+      expect(page).to have_no_field('collection')
+    end
+  end # with admin user
+
+  context 'with viewing user' do
+    let(:druid) { 'druid:hv992ry2431' }
+    let(:item) do
+      item = instantiate_fixture(druid, Dor::AdminPolicyObject)
+      # allow(Dor).to receive(:find).with(@druid).and_return(@item)
+      # allow(Dor::Item).to receive(:find).with(@druid).and_return(@item)
+      item
+    end
+    before :each do
+      # use a viewing user that depends on workgroup roles for access
+      view_user
+      allow(@view_user).to receive(:is_admin).and_return(false)
+      allow(@view_user).to receive(:is_manager).and_return(false)
+      allow(@view_user).to receive(:is_viewer).and_return(false)
+    end
+
+    it 'allows a user of an authorized workgroup to view an APO' do
+      # Ensure all user authority methods return false and the
+      # user belongs to at least one authorized workgroup.
+      dor_viewer_roles = ["dor-viewer", "sdr-viewer"]
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.